examples/streamlit_app/app.py

```diff
@@ -0,0 +1,245 @@
+
+        redirect_url = st.text_input("Enter the redirect URL:")
+
+        if redirect_url:
+            if "?code=" not in redirect_url:
+                st.error("Invalid redirect URL. Please make sure you copied the entire URL.")
+                return None, None
+
+            auth_code = redirect_url.split("?code=")[1]
+
+            try:
+                access_token = get_enphase_access_token(auth_code)
+                st.session_state['access_token'] = access_token
+                return access_token, os.getenv('ENPHASE_SYSTEM_ID')
+            except Exception as e:
+                st.error(f"Error getting access token: {str(e)}")
+                return None, None
+    else:
+        return st.session_state['access_token'], os.getenv('ENPHASE_SYSTEM_ID')
+
+    return None, None
+
+def make_pv_data(site: PVSite, ts: pd.Timestamp, access_token: str = None, enphase_system_id: str = None) -> xr.Dataset:
+    live_generation_kw = None
+
+    if site.inverter_type == 'enphase' and access_token and enphase_system_id:
+        live_generation_kw = get_enphase_data(enphase_system_id, access_token)
+
+    da = process_pv_data(live_generation_kw, ts, site)
+    return da
+
+def predict_ocf(site: PVSite, model=None, ts: datetime | str = None, nwp_source: str = "icon", access_token: str = None, enphase_system_id: str = None):
+    if ts is None:
+        ts = pd.Timestamp.now().round("15min")
+    if isinstance(ts, str):
+        ts = datetime.fromisoformat(ts)
+
+    nwp_xr = get_nwp(site=site, ts=ts, nwp_source=nwp_source)
+    pv_xr = make_pv_data(site=site, ts=ts, access_token=access_token, enphase_system_id=enphase_system_id)
+
+    pred_df = forecast_v1_tilt_orientation(nwp_source, nwp_xr, pv_xr, ts, model=model)
+    return pred_df
+
+def run_forecast(site: PVSite, model: str = "gb", ts: datetime | str = None, nwp_source: str = "icon", access_token: str = None, enphase_system_id: str = None) -> pd.DataFrame:
+    if model == "gb":
+        return predict_ocf(site, None, ts, nwp_source, access_token, enphase_system_id)
+    elif model == "xgb":
+        return predict_tryolabs(site, ts)
+    else:  
+        raise ValueError(f"Unsupported model: {model}. Choose between 'xgb' and 'gb'")
+
+def fetch_data_and_run_forecast(access_token: str = None, enphase_system_id: str = None):
+    with st.spinner("Running forecast..."):
+        try:
+            timestamp = datetime.now().timestamp()
+            timestamp_str = datetime.fromtimestamp(timestamp, tz=timezone.utc).strftime('%Y-%m-%d %H:%M:%S')
+            ts = pd.to_datetime(timestamp_str)
+
+            site_live = PVSite(latitude=51.75, longitude=-1.25, capacity_kwp=1.25, inverter_type="enphase")
+            site_no_live = PVSite(latitude=51.75, longitude=-1.25, capacity_kwp=1.25)
+
+            predictions_with_recent_pv_df = run_forecast(site=site_live, ts=ts, access_token=access_token, enphase_system_id=enphase_system_id)
```

Comment:
has run foreacst been updated to take `access_token` and `enphase_system_id`

Reply:
> has run foreacst been updated to take `access_token` and `enphase_system_id`

ive modified the run_forecast function in the app.py of streamlit, so that it doesnt use the CLI based functions